sanic/sanic.py

```diff
@@ -54,6 +56,18 @@ def __init__(self, name=None, router=None,
     # -------------------------------------------------------------------- #
 
     # Decorator
+    def listener(self, event):
+        """
+        Create a listener from a decorated function.
+
+        :param event: Event to listen to.
+        """
+        def decorator(listener):
+            self.listeners[event].append(listener)
```

Comment:
Can you make it so that listener method accepts either `before_server_start` or `before_start` as synonyms, etc...? There are a few ways you could do this (e.g. with a dict like {'before_start': 'before_server_start'}, then get the original thing as a default value, or some logic based on the string, where you insert `server` if it's missing)